### DISCUSSION TITLE
Wiring DS18B20 thermometers directly to a Raspberry Pi

### DISCUSSION THREAD

**Discussion by delight-f** [thorrak/fermentrack#577] [Q&A] [answered]
Might be a silly question, but is there support for wiring temperature sensors directly to a Raspberry Pi running Fermentrack, or does it have to be run through a BrewPi Controller e.g. an ESP8266 that then interfaces with a Pi running Fermentrack?

**Comment by vanosg** [comment, marked as answer]
Not a silly question at all - I tried rewriting BrewPi several years ago to do just that, but failed :)  The short answer is no (with Fermentrack). But here's why:

An ESP8266 is an embedded device that is only running the code for Fermentrack- nothing else. A Raspberry Pi would have that code, plus everything else it needs to function as an OS. That means, while not necessarily likely, there is a lot of code that can could hypothetically cause issues on that device and result in it crashing or otherwise freezing (and I'm a firm believer in Murphy. Were that to happen and your heater had been in the 'on' position, or maybe it was all in the off position but you're brewing in your cold garage, your batch would very possibly be ruined. in my seven years of brewing with some variant of BrewPi, I'd say I've come home to discover a locked up Pi probably twice. Not a huge number, but still that's potentially two batches of ruined beer. The good news is, the ESP8266 is like $10 on Amazon so hopefully it would be a worthwhile investment for you.

That said, I think there is another distro of BrewPi that @lbussy maintains (I may be wrong on this) that eliminates the hardware controller, but I can't be certain on that. BrewPi Remix? BrewPi Lite? Try googling those terms and see if something comes back.

But my suggestion, as someone who was in your shoes not too long ago, is to go with the ESP8266. Good luck, and check back in with any other questions. Hope that helps!

**Comment by thorrak** [comment]
I'm pretty sure the project you're thinking of is [Fuscus](https://github.com/andrewerrington/fuscus). That project is actually what got me involved in the BrewPi ecosystem - but you're right, the benefits of a dedicated controller are significant enough that I would recommend going down that route if possible.

Fuscus is supported in Fermentrack, however, though you will need to go through the "advanced setup" workflow as it requires the use of UNIX sockets to connect with Fermentrack's BrewPi-Script.

**Comment by ericpare1** [comment]
Is it possible to get the probe temperature (I'm guessing with some custom code) ? I'm able to get the value from /sys/bus/w1/devices/ and I wanted to display it in the dashboard :
![image](https://user-images.githubusercontent.com/60197654/105122569-73457400-5aa4-11eb-93d1-14352c12c514.png)

**Reply by thorrak** [reply]
My recommendation would be to just set up Fuscus. It should provide the interface you want (a BrewPi interface to sensors connected to an RPi)
